src/server/lib/okta/idx/shared/submitPasscode.ts

```diff
@@ -221,3 +222,72 @@ export const setPasswordAndRedirect = async ({
 	// redirect the user to set a global session and then back to completing the authorization flow
 	return expressRes.redirect(303, loginRedirectUrl);
 };
+
+/**
+ * @name skipPasswordEnrollment
+ * @description Okta IDX API/Interaction Code flow - After email factor verification for a new account, we can choose to skip setting a password for the user, and set the emailValidated flag in Okta, and complete the authorization code flow.
+ * @param stateHandle - The state handle from the previous step
+ * @param expressReq - The express request object
+ * @param expressRes - The express response object
+ * @param ip - The ip address
+ * @returns Promise<void> - Performs a express redirect
+ */
+export const skipPasswordEnrollment = async ({
+	stateHandle,
+	expressReq,
+	expressRes,
+	ip,
+}: {
+	stateHandle: IdxBaseResponse['stateHandle'];
+	expressReq: Request;
+	expressRes: ResponseWithRequestState;
+	ip?: string;
+}): Promise<void> => {
+	const skipResponse = await skip(stateHandle, ip);
+
+	if (!isChallengeAnswerCompleteLoginResponse(skipResponse)) {
+		throw new OAuthError({
+			error: 'idx.invalid.response',
+			error_description: 'The response was not a complete login response',
+		});
+	}
+
+	const loginRedirectUrl = getLoginRedirectUrl(skipResponse);
+
+	// set the validation flags in Okta
+	const { id } = skipResponse.user.value;
+	if (id) {
+		// we only want to set the emailValidated flag here,
+		// as the password is not set yet
+		await validateEmailAndPasswordSetSecurely({
+			id,
+			ip,
+			updateEmail: true,
+			updatePassword: false,
+		});
+	} else {
+		logger.error(
+			'Failed to set validation flags in Okta as there was no id',
+			undefined,
+		);
```

Comment:
Should we throw an exception here?

Reply:
hmm.. I implemented this in the similar way to how the `submitPassword`/`submitPasscode` functions were set up in this same file, which don't throw either.

But i'm wondering if this error is even relevant, it should'nt even be possible to get here if a user didn't have an Okta id!

Reply:
Yeah I've just checked the code and we don't need it! `id` is defined! If `id` is missing it would throw automatically as the zod parsing would've failed.

Reply:
Have removed the `if` check.